docs/source/federation/metrics.md

```diff
@@ -0,0 +1,37 @@
+
+## Turning it on
+
+Ensure that all dependencies on `apollo-server` are at version `2.7.0` or higher. Provide an API key to your gateway via the `ENGINE_API_KEY` environment variable for the gateway to report metrics to the default ingress. To ensure that implementing services do not report metrics as well, do not provide them with an `ENGINE_API_KEY` or set `{ engine: true, reporting: false }` in the constructor options to ApolloServer. These options will cause the Apollo gateway to collect tracing information from the underlying federated services and pass them on, along with the query plan, to the Apollo metrics ingress. To change which endpoint you would like to report the tracing information to, you can modify the `endpointUrl` in the `engine` key of the constructor to Apollo Server. Currently, only Apollo Server supports detailed metrics insights as an implementing service, but we would love to work with you to implement the protocol in other languages!
+
+> NOTE: By default, metrics will be reported to the `'current'` variant. To change the variant for reporting, set the `ENGINE_GRAPH_VARIANT` environment variable.
+
+## How tracing data is exposed from a federated service
+
+The Apollo gateway looks to the `extensions` field of all service responses for the presence of an `ftv1` field. This field contains a representation of the tracing information for the sub-query that was executed against the service, sent as the base64 encoding of the [protobuf representation](https://github.com/apollographql/apollo-server/blob/master/packages/apollo-engine-reporting-protobuf/src/reports.proto) of the trace. By default, a federated Apollo Server service will emit this tracing information in its extension if sent the header pair `'apollo-federation-include-trace': 'ftv1'`. By setting the gateway's API key via the environment variable, it will send this header pair by default as well.
+
+## How the gateway constructs and reports traces
+
+The Apollo gateway constructs traces in the shape of the [query plan](https://www.apollographql.com/docs/apollo-server/federation/implementing/#inspecting-query-plans), embedding an individual `Trace` at each fetch node to indicate the sub-query traces and in which order they were fetched from the underlying services. The field-level statistics that the metrics ingress aggregates from these traces are collected over the fields over which the operation was executed **in the implementing services**. In other words, field stats are collected based on the operations the query planner makes rather than the operations that the clients make. On the other hand, query-level statistics are aggregated over the operations exected **by the client**, which means that even if query-planning changes, statistics will still be corresponded to the same client-delivered operation.
```

Comment:
This is great to have written out but seems like more information than we've ever given about traces before and I'm not sure about the relatability/digestibility of some of the more technical aspects of these bits to the end-user.  e.g. the bit about _`Trace`s embedded at fetch nodes._, or the _metrics ingress_ (maybe this should just say _the Apollo Platform_?).

Reply:
Yeah this section of docs was definitely intended to be for very advanced users. I was hoping that I could start to use this section to build out other users' interest in adding tracing support to other language's federation support. I think I'll keep it and make the wording less precise.